solutions/gold/cf-585D.mdx

```diff
@@ -0,0 +1,138 @@
+---
+id: cf-585D
+source: CF
+title: Lizard Era Beginning
+author: Justin Ji
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/20898)
+
+<Spoiler title = "Hint 1">
+
+Choosing two companions per task is equivalent to excluding one companion
+per task.
+
+</Spoiler>
+
+<Spoiler title = "Hint 2">
+
+Let's say we have a pair $(a, b)$ and $(c, d)$. If we want $a + c$ to equal $b + d$,
+then $a - b$ must be equal to $d - c$. How can this help us perform MITM on the
+array?
+
+</Spoiler>
+
+<Spoiler title = "Explanation">
+
+## Explanation
+
+With the information given in the hints, we can do MITM and consider every possible
+way to choose companions to exclude in our left and right subarrays.
+
+Given the information in Hint 2, we can efficiently join results from the
+left and right sides. In a valid answer, the adjacent attitudes from the left side
+must be the opposite of the adjacent differences from the right side.
+With this observation, we can find the best pairing for each possible
+result from the left side in $\mathcal{O}(\log{N})$.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(3^{\left \lceil{\frac{N}{2}}\right \rceil}
+\log{3^{\left \lceil{\frac{N}{2}}\right \rceil}})$
```

Comment:
are ceils needed

Reply:
the complexity is more accurate if you leave the ceils in

Reply:
![image](https://github.com/user-attachments/assets/89ab604e-0430-4a14-a4c1-327221981fd5)

maybe uh wrap the log term in parents and use \left( and \right)

Reply:
done, lmk if i need to do more changes